### PR TITLE
[DPE-3562] Don't block on failure to get the db version

### PR DESCRIPTION
**Pull request by dragomirp**
Conditionally update the db version field in legacy relations if database is not accessible.

Related to #352 and #566

### CODE REVIEW

File: src/relations/db.py [hunk 1]
```diff
@@ -392,6 +391,8 @@ def update_endpoints(self, relation: Relation = None) -> None:
                 "state": self._get_state(),
                 "extensions": ",".join(required_extensions),
             }
+            if postgresql_version:
+                data["version"] = postgresql_version
```
Comment (dragomirp) [inline src/relations/db.py:395]: Only add the version if we were able to get it.

File: src/relations/db.py [hunk 2]
```diff
@@ -319,14 +319,14 @@ def update_endpoints(self, relation: Relation = None) -> None:
         if len(relations) == 0:
             return
 
+        postgresql_version = None
         try:
             postgresql_version = self.charm.postgresql.get_postgresql_version()
```
Comment (dragomirp) [inline src/relations/db.py:324]: Alternatively we can use `cluster.get_postgresql_version` instead and get the version from the snap.

File: lib/charms/postgresql_k8s/v0/postgresql.py [hunk 3]
```diff
@@ -425,14 +425,20 @@ def get_postgresql_timezones(self) -> Set[str]:
...
+    def get_postgresql_version(self, current_host=True) -> str:
         """Returns the PostgreSQL version.
 
         Returns:
             PostgreSQL version number.
         """
+        if current_host:
+            host = self.current_host
+        else:
+            host = None
```
Comment (dragomirp) [inline lib/charms/postgresql_k8s/v0/postgresql.py:437]: Use current_host by default, I think we still need the primary for pgbouncer.

**Review by taurus-forever — APPROVED** [review]
LGTM! Tnx!

**Review by marceloneppel — APPROVED** [review]
Thanks!

**Pull request merged**